src/coresyftools/coresyftools/gpt_tool.py

```diff
@@ -55,20 +63,24 @@ def run(self, bindings):
         bindings = bindings.copy()
         operator = self.operation.get('operation')
         if 'graph' in self.operation and self.operation['graph']:
-            graph_file = os.path.join(self.context_directory, self.DEFAULT_GPT_GRAPH_FILE_NAME)
+            graph_file = os.path.join(self.context_directory,
+                                      self.DEFAULT_GPT_GRAPH_FILE_NAME)
             if not os.path.exists(graph_file):
                 raise GPTGraphFileNotFound(graph_file)
             operator = graph_file
         if 'parameters' in self.operation:
             bindings.update(self.operation['parameters'])
         source = bindings.pop(self.arg_parser.inputs[0])
+        source_with_ext = self._add_snap_file_extension(source)
```

Comment:
This is a little confusing because the extension can be added or not and it's not clear why and when it is added. But I do not have any suggestion, let's leave it this way for now.